tests/helpers/yadda-annotations.js

```diff
@@ -56,4 +45,35 @@ function runScenario(featureAnnotations, scenarioAnnotations) {
   return checkAnnotations(scenarioAnnotations);
 }
 
-export { runFeature, runScenario };
+// setup tests
+// you can override these function to add additional setup setups, or handle new setup related annotations
+function setupFeature(featureAnnotations) {
+  return setupYaddaTest(featureAnnotations);
+}
+
+function setupScenario(featureAnnotations, scenarioAnnotations) {
+  let setupFn = setupYaddaTest(scenarioAnnotations);
+  if (setupFn && (featureAnnotations.application || featureAnnotations.rendering || featureAnnotations.context)) {
+    throw new Error('You must not assign any @application, @rendering or @context annotations to a scenario as well as its feature!');
+  }
+  return setupFn;
+}
+
+function setupYaddaTest(annotations) {
+  if (annotations.application) {
+    return setupApplicationTest;
+  }
+  if (annotations.rendering) {
+    return setupRenderingTest;
+  }
+  if (annotations.context) {
+    return setupTest;
+  }
+}
```

Comment:
Can this do setupApplicationTest by default if rendering or context are not set?  I'm assuming application tests are most commonly used with BDD testing.

Reply:
Hm, I though about this as well. But by this you would loose the ability to run plain tests without any Ember specific setup. This might be useful if you want to test some utility functions (pure functions), or some business logic classes which are not coupled to Ember and thus don't require special setup work. Like true isolated unit tests.

I certainly agree that it's mostly application tests in practice (same for us). But I think it would be nice to allow any test setup by default. Maybe we can describe this as an example in the Readme for how to customize the annotations file, e.g. how to run all test as an application test without any required tagging in your .feature files?